From 987892128d1e83cabbd8c4a436acc56c99ecc8a3 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 19 Apr 2024 13:42:11 +0200
Subject: [PATCH 0803/1695] dts: stm32mp15: harden RCC secure configuration on
 ST boards

Enable STM32MP15 RCC secure hardening configuration on ST boards
to assign SoC clocks, reset controllers and PWR regulators to
OP-TEE secure world.

Align internal board to commit 08278885852f ("dts: stm32mp15: harden RCC
secure configuration on ST boards").

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c419aa5369e05671c8b5363a3a6465a646938dd3 dts: stm32mp15: harden RCC secure configuration on ST boards
00324efc268d2b3e88b030c049aca03fe0688d58 dts: stm32mp15: harden RCC secure configuration on ST boards]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I6c7459ff75db91bdf1a57cbc21311ef10b66d36d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/375269
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp157a-ed1.dts  | 2 ++
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 1 -
 core/arch/arm/dts/stm32mp157d-ed1.dts  | 2 ++
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 1 -
 4 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp157a-ed1.dts b/core/arch/arm/dts/stm32mp157a-ed1.dts
index f1b641217..a54b5dad4 100644
--- a/core/arch/arm/dts/stm32mp157a-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157a-ed1.dts
@@ -248,6 +248,8 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure";
+
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 034dbbe47..1443b25d1 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -408,7 +408,6 @@
 
 &rcc {
 	compatible = "st,stm32mp1-rcc-secure";
-	status = "okay";
 
 	st,clksrc = <
 		CLK_MPU_PLL1P
diff --git a/core/arch/arm/dts/stm32mp157d-ed1.dts b/core/arch/arm/dts/stm32mp157d-ed1.dts
index 6d968ae1f..8a2ea5d5b 100644
--- a/core/arch/arm/dts/stm32mp157d-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157d-ed1.dts
@@ -247,6 +247,8 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure";
+
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 7c3dbe2da..d46a4574c 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -545,7 +545,6 @@
 
 &rcc {
 	compatible = "st,stm32mp1-rcc-secure";
-	status = "okay";
 
 	st,clksrc = <
 		CLK_MPU_PLL1P
-- 
2.47.2

